sidecar/app/query/base.py

```diff
@@ -77,23 +83,55 @@ def get_from_query_id(cls, query_id) -> Optional["Query"]:
             if query:
                 return query
             raise e
-        if query.status_file_path.exists():
-            with query.status_file_path.open("r") as f:
-                status_str = f.readline()
-                query.status = Status[status_str]
-                return query
-        return None
+        query.load_history_from_file()
+        if query.status == Status.UNKNOWN:
+            return None
+        return query
+
+    def load_history_from_file(self):
+        if self.status_file_path.exists():
+            self.logger.debug(
+                f"Loading query {self.query_id} status history "
+                f"from file {self.status_file_path}"
+            )
+            with self.status_file_path.open("r") as f:
+                for line in f:
+                    status_str, timestamp = line.split(",")
+                    self._status_history.append(
+                        StatusChangeEvent(
+                            status=Status[status_str], timestamp=float(timestamp)
+                        )
+                    )
+
+    @property
+    def _last_status_event(self):
```

Comment:
is it expected to be public with `_` or @property does not work as I think it works?

Reply:
this is intended to be private. `@property` is just a helper that makes `instance.method` a shorthand for `instance.method()`.